consensus/core/src/authority_node.rs

```diff
@@ -86,13 +87,23 @@ where
             context.clone(),
             commit_consumer.sender,
             commit_consumer.last_processed_index,
-            dag_state,
+            dag_state.clone(),
             store.clone(),
         );
+
+        // TODO: Move this into core new when dag state is the interface for all store reads/writes
+        let last_decided_leader = dag_state.read().last_commit_leader();
```

Comment:
Why not move dag_state into Core now?

Reply:
Yeah I would encourage you @arun-koshy as well injecting the `DagState` in Core to motivate us on making the `DagState` the main access point for our data